docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,624 @@
+use later during the evaluation step.
+
+Though each of the techniques above - T5, OpenAI, and RAGAS - confers certain advantages and disadvantages, all of them let you build a good evaluation dataset including reasonably good questions and ground-truths to evaluate your RAG system. Which you decide to use depends on the specifics of your use case.
+
+### Tuning with answers
+
+In each tuning cycle (i.e., adjusting model parameters, training the model, evaluating the model), your evaluation dataset should be tuned using RAG pipeline-generated `answer`s. This requires developing a subroutine that facilitates the construction of an evaluation dataset in the expected RAGAS format (`question`, `answer`, `contexts`, `ground_truths` - as indicated in the above code snippet) executing the provided questions through your RAG system.
+
+Such a subroutine might look like this:
+
+```python
+## Prepare the evaluation dataset to evaluate our RAG system
+from datasets import Dataset
+
+# RAGAS Expect ['question', 'answer', 'contexts', 'ground_truths'] format
+'''
+{
+    "question": ['What is quantization?', ...], 
+    "answer": [], ## answer
+    "contexts": [], ## context
+    "ground_truths": [] ## answer expected
+}
+'''
+# loading the eval dataset from HF
+qdrant_qa = load_dataset("atitaarora/qdrant_doc_qna", split="train")
+
+questions = []
+answers = []
+contexts = []
+ground_truths = []
+
+for query in qdrant_qa:
+    question = query["question"]
+    questions.append(question)
+    ground_truths.append([query["answer"]])
+    
+    rag_response = query_with_context(question)
+    answers.append(rag_response)
+    search_result_context = client.query(collection_name=COLLECTION_NAME, query_text=question,limit=4)
+    context = ["document:"+r.document+",source:"+r.metadata['source'] for r in search_result_context]
+    contexts.append(context)    
+
+rag_response_data = {
+    "question": questions,
+    "answer": answers,
+    "contexts": contexts,
+    "ground_truths": ground_truths
+}
+
+## This dataset 'rag_response_dataset', is created to evaluate the RAG system using the RAGAS format.
+## Note: this dataset is specifically for evaluation purposes. Therefore, it needs to be recreated every time changes are made to the RAG config.
+rag_response_dataset = Dataset.from_dict(rag_response_data)
+# The dataset is then exported as a CSV file, with a filename that includes details of the experiment for easy identification, such as the chunk size used in this case -  rag_response_dataset.to_csv('rag_chunk_512.csv')
+```
+
+The subroutine above uses an abstraction of the RAG pipeline method `query_with_context()`. The `rag_response_dataset` uses an older format in which `ground_truths` is a list of strings, while the latest format uses `ground_truth` as a single string value (representing the correct answer to a question). 
+We will show a sample naive-rag example in a bit.
```

Comment:
how is the sample naive-RAG relevant here in this paragraph?
which methods are used there?
try to present a logic for the user choosing between formats... it's not clear why one would choose one or the other here.

Reply:
@atarora 
Still not sure why "sample naive RAG" is mentioned here. Pls explain here (I will make all remaining edits to the PR).

Reply:
Also, is "sample" supposed to be "simple"?

Reply:
Please refer to : https://www.promptingguide.ai/research/rag#rag-paradigms
Which I agree has not been part of source in this article but has been a very standard terminology in the generative AI space and what I intend to tell the reader - that we are building 'naive kind of rag' 
I wonder if you have a suggestion handling this better?

Reply:
@atarora 
Is the fact that you're building a naive RAG system related to the preceding and following sentences in this paragraph? 
In other words, the fact that you're building naive RAG affects your choice of RAGAS format? your format for `ground_truth`?

Reply:
@robertdhayanturner 
1. The format of ground_truth isn't affected by type of RAG but this statement here is a disclosure for reader that method `query_with_context()` is an abstraction of our RAG pipeline and we're building naive rag or simple rag in this example notebook as there are plenty of different kind.
It is a good idea for readers to know what kind of RAG is being built in the example as that could be of an individual interest to the readers. Another reason it is explicitly mentioned here as I am going to be using a 'hybrid' rag in my next article.

2. Simple rag and Naive rag means the same , we are showcasing an example of naive rag (simple rag) and hence I am using the word `sample` , although I don't feel the statement `sample naive rag` needed discussion , I can drop it if it helps at the risk of people a). not being aware of what kind of rag are we evaluating b). not being aware that it is only a `sample` application aka manifestation of a production grade app.